CDCgov/prime-reportstream
# Created a method that allows you to look up the LOINC code for a FHIR…

… observation that reuses the existing LivdMappers code

This PR ...

**If you are suggesting a fix for a currently exploitable issue, please disclose the issue to the prime-reportstream team directly outside of GitHub instead of filing a PR, so we may immediately patch the affected systems before a disclosure. See [SECURITY.md/Reporting a Vulnerability](https://github.com/CDCgov/prime-reportstream/blob/master/SECURITY.md#reporting-a-vulnerability) for more information.**

Test Steps:
1. I will attach a bundle to use to test. Use this as the bundle for FHIR Path. Run the Run CLI FHIR  Patch Sample Data Environment Configuration changing the file to something like below that points to the attached file
     ex. src/testIntegration/resources/datatests/HL7_to_FHIR/bundle1.txt
2. In FHIR Path set the resource to resource=Bundle.entry.resource.ofType(Observation)[0]
3. Run %resource.livdTableLookup("Test Performed LOINC Code")
4. Make sure that a LOINC code is returned

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue 7949

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue
- https://github.com/CDCgov/prime-reportstream/issues/8434
- https://github.com/CDCgov/prime-reportstream/issues/8435

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints? No
    - new endpoint A No
    - new endpoint B No
- Does this PR include changes in authentication and/or authorization of existing endpoints? No
- Does this change introduce new dependencies that need vetting? No
- Does this change require changes to our infrastructure? No
- Does logging contain sensitive data? No
- Does this PR include or remove any sensitive information itself? No

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
